1a.cpp

```diff
@@ -0,0 +1,56 @@
+#include <iostream>
+#include <vector> 
+using namespace std;
+
+int main()
+{
+	cout << "Введите размер вектора\n";
+	int vecsize;
+	cin >> vecsize;
+	if (cin.fail())
+	{
+		cout << "Введено нецелочисленное значение.\n";
+		return 0;
+	}
+	else if (vecsize <= 0)
+	{
+		do
+		{		
+			cout << "Размер вектора должен быть положителен. Введите значение заново.\n";
+			cin >> vecsize;
+		}
+		while (vecsize <= 0);
+	}
+	vector<int> v1;
+	cout << "Коллекция чисел: \n";
+	for(int counter = 0, i = 0; i < vecsize; counter++, i++)
```

Comment:
As a rule of thumb, prefix increment is always better than postfix if it is not a composite expression.